libfwupd/fwupd-self-test.c

```diff
@@ -121,6 +121,39 @@ fwupd_remote_download_func (void)
 			 LOCALSTATEDIR "/lib/fwupd/remotes.d/lvfs/metadata.xml.gz.asc");
 }
 
+/* verify we used the FirmwareBaseURI just for firmware */
+static void
+fwupd_remote_baseuri_func (void)
+{
+	gboolean ret;
+	g_autofree gchar *firmware_uri_str = NULL;
+	g_autofree gchar *fn = NULL;
+	g_autoptr(FwupdRemote) remote = NULL;
+	g_autoptr(GError) error = NULL;
+	g_autoptr(SoupURI) firmware_uri = NULL;
+
+	remote = fwupd_remote_new ();
+	fn = g_build_filename (TESTDATADIR, "tests", "firmware-base-uri.conf", NULL);
+	ret = fwupd_remote_load_from_filename (remote, fn, NULL, &error);
+	g_assert_no_error (error);
+	g_assert (ret);
+	g_assert_cmpint (fwupd_remote_get_kind (remote), ==, FWUPD_REMOTE_KIND_DOWNLOAD);
+	g_assert_cmpint (fwupd_remote_get_keyring_kind (remote), ==, FWUPD_KEYRING_KIND_GPG);
+	g_assert_cmpint (fwupd_remote_get_priority (remote), ==, 0);
+	g_assert (fwupd_remote_get_enabled (remote));
+	g_assert_cmpstr (fwupd_remote_get_metadata_uri (remote), ==,
+			 "https://s3.amazonaws.com/lvfsbucket/downloads/firmware.xml.gz");
+	g_assert_cmpstr (fwupd_remote_get_metadata_uri_sig (remote), ==,
+			 "https://s3.amazonaws.com/lvfsbucket/downloads/firmware.xml.gz.asc");
+
+	firmware_uri = fwupd_remote_build_uri (remote, "http://bbc.co.uk/firmware.cab", &error);
+	g_assert_no_error (error);
+	g_assert (firmware_uri != NULL);
+	firmware_uri_str = soup_uri_to_string (firmware_uri, FALSE);
+	g_assert_cmpstr (firmware_uri_str, ==,
+			 "https://my.fancy.cdn/firmware.cab");
```

Comment:
I think you should instead be checking this with `fwupd_remote_build_firmware_uri `